utils.go

```diff
@@ -226,6 +229,15 @@ func DefaultStoreOptions(rootless bool, rootlessUID int) (StoreOptions, error) {
 			}
 			if storageOpts.GraphRoot == "" {
 				storageOpts.GraphRoot = defaultRootlessGraphRoot
+			} else if storageOpts.RootlessStoragePath != "" {
+				rootlessStoragePath := strings.Replace(storageOpts.RootlessStoragePath, "$HOME", homedir.Get(), -1)
```

Comment:
Oh, ouch. This sort of string replacement isn't as simple as that: for instance, what if someone writes `$HOMEDIR` or `$USERNAME`? What you probably want here is something closer to:
```
    look for dollar sign followed by any number of letters
    if those letters are HOME, UID, or USER, replace as needed
    otherwise, throw an error
```